apps/src/lib/kits/maker/toolkit.js

```diff
@@ -0,0 +1,118 @@
+/*
+ * @file Static interface to Maker Toolkit, to simplify App Lab code interfacing
+ * with maker and provide clean setup/cancel/reset patterns.
+ */
+import codegen from '../../../codegen';
+import {getStore} from '../../../redux';
+import CircuitPlaygroundBoard from './CircuitPlaygroundBoard';
+import * as commands from './commands';
+import * as dropletConfig from './dropletConfig';
+import MakerError, {ConnectionCanceledError} from './MakerError';
+import {findPortWithViableDevice} from './portScanning';
+import * as redux from './redux';
+
+// Re-export some modules so consumers only need this 'toolkit' module
+export {dropletConfig, MakerError};
+
+/**
+ * @type {CircuitPlaygroundBoard} The current board controller, populated when
+ * connected, null when not connected.  There can be only one at any time.
+ */
+let currentBoard = null;
+
+/**
+ * Enable Maker Toolkit for the current level.
+ */
+export function enable() {
+  getStore().dispatch(redux.enable());
+}
+
+/**
+ * @returns {boolean} whether Maker Toolkit is enabled for the current level
+ */
+export function isEnabled() {
+  return redux.isEnabled(getStore().getState());
+}
+
+/**
+ * Called when starting execution of the student app.
+ * Looks for a connected board, sets up an appropriate board controller,
+ * and injects needed references into the interpreter and its commands.
+ * @param {JSInterpreter} interpreter
+ * @param {function} onDisconnect
+ * @return {Promise}
+ *   Resolves (with no value) when connection is successful and the board
+ *   controller is ready to use.
+ *   Resolves immediately if maker toolkit is disabled.
+ *   Rejects with a MakerError when the connection process is cancelled or
+ *   fails in an expected way that we handle gracefully (for example, no board
+ *   plugged in).
+ *   Rejects with a plain Error if something more unexpected happens.
+ */
+export function connect({interpreter, onDisconnect}) {
```

Comment:
Tried to document this well, please let me know if you see something missing or that could be phrased better.  In short:

* Resolves if/when maker is ready to continue with running the app.
* Rejects if anything gets in the way.  There are lots of possible rejection reasons though:
  * `ConnectionCanceledError` if we intentionally aborted the connection (say, clicked "Reset") while trying to connection to the board.
  * `ConnectionFailedError` if we were unable to connect for some known/common reason, like the board not being plugged in.
  * `Error` if something more surprising went wrong.

Note that `ConnectionCanceledError` and `ConnectionFailedError` are both subclasses of `MakerError` - that's important later.